### PR TITLE
Add optional strand-pair averaging to compute_track_means

### PR DESCRIPTION
**Pull request by gtca**
Port scaling semantics from upstream.

**Feature:** add an optional `strand_pair_groups` argument to `compute_track_means` that averages the per-track nonzero means within each (plus, minus) pair so paired strands share a scaling factor — matching AlphaGenome's official scaling semantics.

Our `compute_track_means()` in `src/alphagenome_pytorch/extensions/finetuning/datasets.py` computes per-bigwig nonzero means but does not average plus/minus strand pairs by track name. Upstream notebooks pair them so the two strands of the same assay receive identical scaling.

```python
 def compute_track_means(
     bigwig_files: list[str],
     bed_file: str,
     sequence_length: int = 1_048_576,
     resolution: int = 1,
     max_samples: int | None = None,
+    strand_pair_groups: list[tuple[int, int]] | None = None,
 ) -> torch.Tensor:
     ...
+    if strand_pair_groups is not None:
+        for plus_idx, minus_idx in strand_pair_groups:
+            paired_mean = 0.5 * (track_means[plus_idx] + track_means[minus_idx])
+            track_means[plus_idx] = paired_mean
+            track_means[minus_idx] = paired_mean
```

Default `None` keeps existing behavior.

Usage:

```python
track_means = compute_track_means(
    bigwig_files=['atac_plus.bw', 'atac_minus.bw', 'rna_plus.bw', 'rna_minus.bw'],
    bed_file='train.bed',
    strand_pair_groups=[(0, 1), (2, 3)],   # (plus_idx, minus_idx) per assay
)
# track_means[0] == track_means[1]; track_means[2] == track_means[3]
```

### CODE REVIEW

**Comment by gtca** [comment]
### Strand-pair track-mean averaging for stranded fine-tuning

#### What's supported now

**CLI — `auto` (pair consecutive bigwigs `(0,1),(2,3),…`; order them `+ - + -`):**
```bash
python scripts/finetune.py --mode lora \
    --genome hg38.fa --pretrained-weights alphagenome.pth \
    --train-bed train.bed --val-bed val.bed \
    --modality rna_seq --bigwig rna_plus.bw rna_minus.bw \
    --strand-pairs "rna_seq:auto"
```

**CLI — explicit indices (when bigwigs aren't interleaved):**
```bash
    --modality rna_seq --bigwig rp1.bw rp2.bw rm1.bw rm2.bw \
    --strand-pairs "rna_seq:0,2;1,3"
```

**Mixed stranded + unstranded modalities** — indices are 0-based **per modality**, so unstranded modalities just get no entry and are untouched:
```bash
    --modality atac    --bigwig atac1.bw atac2.bw \
    --modality rna_seq --bigwig rna_plus.bw rna_minus.bw \
    --strand-pairs "rna_seq:auto"          # atac unaffected
```

**Config (`--config`)** — per-modality `strand_pairs`, as `auto` or explicit pairs; CLI `--strand-pairs` overrides config:
```yaml
modalities:
  atac:
    bigwig: [atac1.bw, atac2.bw]
  rna_seq:
    bigwig: [rna_plus.bw, rna_minus.bw]
    strand_pairs: auto          # or [[0, 1], [2, 3]]
```

#### Scope / semantics
- Affects **only** the scaling factor — strands still get independent predictions and loss; they just share one normalization constant.
- `auto` requires an even bigwig count for that modality and pairs them positionally.

#### Hardening
- `compute_track_means` now validates `strand_pair_groups`: in-range, two distinct indices, and no index reused across pairs — raising a clear `ValueError` instead of silently wrapping a negative index or producing order-dependent results.
- Tests switched from leaking `tempfile.mkdtemp()` to pytest `tmp_path`.

#### Tests
- `tests/unit/test_track_means.py`: default-no-pairing parity, averaging, and 4 invalid-input cases.
- `tests/unit/test_finetune_multitask.py`: `auto`, explicit string, none→`None`, odd-count rejection, unknown-modality rejection, config list-of-lists, CLI-overrides-config.

#### Docs
- `finetuning/README.md`: CLI example, YAML example, and notes updated.
